remote-api/src/main/java/org/alfresco/rest/api/impl/SizeDetailsImpl.java

```diff
@@ -0,0 +1,212 @@
+    private static final String STATUS = "status";
+    private static final String ACTIONID = "actionId";
+    private static final String INVALID_NODEID = "Invalid parameter: value of nodeId is invalid";
+    private static final String INVALID_JOBID = "Invalid parameter: value of jobId is invalid";
+    private static final String FOLDER = "folder";
+    private Nodes nodes;
+    private NodeService nodeService;
+    private PermissionService permissionService;
+    private ActionService actionService;
+    private SimpleCache<Serializable,Map<String, Object>> simpleCache;
+    private int defaultItems;
+
+    public void setNodes(Nodes nodes)
+    {
+        this.nodes = nodes;
+    }
+
+    public void setPermissionService(PermissionService permissionService)
+    {
+        this.permissionService = permissionService;
+    }
+
+    public void setNodeService(NodeService nodeService)
+    {
+        this.nodeService = nodeService;
+    }
+
+    public void setActionService(ActionService actionService)
+    {
+        this.actionService = actionService;
+    }
+
+    public void setSimpleCache(SimpleCache<Serializable, Map<String, Object>> simpleCache)
+    {
+        this.simpleCache = simpleCache;
+    }
+
+    public void setDefaultItems(int defaultItems)
+    {
+        this.defaultItems = defaultItems;
+    }
+
+    /**
+     * generateNodeSizeDetailsRequest : providing HTTP STATUS 202 with jobId.
+     */
+    @Override
+    public NodeSizeDetails generateNodeSizeDetailsRequest(String nodeId)
+    {
+        NodeRef nodeRef = nodes.validateNode(nodeId);
+        validateType(nodeRef);
+        String actionId;
+        if(simpleCache.get(nodeId) == null)
+        {
+            actionId = executeAction(nodeRef, defaultItems, simpleCache);
+        } else
+        {
+            Map<String, Object> result = simpleCache.get(nodeRef.getId());
+            actionId = (String)result.get(ACTIONID);
+        }
+        return new NodeSizeDetails(actionId);
+    }
+
+    /**
+     * getNodeSizeDetails : providing HTTP STATUS 200 with NodeSizeDetails data from cache.
+     */
+    @Override
+    public NodeSizeDetails getNodeSizeDetails(final String nodeId, final String jobId)
+    {
+        NodeRef nodeRef = nodes.validateNode(nodeId);
+        validateType(nodeRef);
+
+        if(simpleCache.get(nodeId) == null)
+        {
+            return new NodeSizeDetails(nodeId, NOT_INITIATED.name());
+        }
+
+        LOG.debug("Executing executorResultToSizeDetail  method");
+        return executorResultToSizeDetail(simpleCache.get(nodeId), nodeId, jobId);
+    }
+
+    /**
+     * Executing Action Asynchronously.
+     */
+    private String executeAction(NodeRef nodeRef, int defaultItems, SimpleCache<Serializable, Map<String, Object>> simpleCache)
+    {
+        Map<String, Object > currentStatus = new HashMap<>();
+        currentStatus.put(STATUS,IN_PROGRESS.name());
```

Comment:
Setting the status to `IN_PROGRESS` immediately upon creating the action is misleading because the actual calculation hasn’t started; we are merely submitting the task to the executor. A status of `QUEUED` or `PENDING` would be more accurate and appropriate until the calculation begins.